eclipse-che/che-plugin-registry
# feat: provide icons for the editors

### What does this PR do?
Add icons for editors generated as a part of plugin registry image.

-  The icons are available by the link: **PLUGIN_REGISTRY_URL/v3/images/**
-  index.json that contains information about each editor, provides an information about the related icon, for example:
```json
{
"id": "cdr/code-server/latest",
"description": "An open source distribution of Visual Studio Code (code-server) for Eclipse Che",
"displayName": "VS Code via code-server",
"links": {
"devfile": "/v3/plugins/cdr/code-server/latest/devfile.yaml"
},
"name": "code-server",
"publisher": "cdr",
"type": "Che Editor",
"version": "latest",
"icon": "/v3/images/vscode.svg"
},
```

In case when the editor's definition doesn't provide an information about icon, the default one will be used: https://pr-check-1885-che-plugin-registry.surge.sh/v3/images/default.svg


### Screenshot/screencast of this PR
![screenshot-pr-check-1885-che-plugin-registry surge sh-2024 02 07-13_06_09](https://github.com/eclipse-che/che-plugin-registry/assets/1271546/71be7561-de63-4bc2-8a53-654c9720d2c5)

![screenshot-pr-check-1885-che-plugin-registry surge sh-2024 02 07-13_08_04](https://github.com/eclipse-che/che-plugin-registry/assets/1271546/85892546-2183-47fe-814f-4cc38a9d2a69)


### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22766

### How to test this PR?


- Open the link https://pr-check-1885-che-plugin-registry.surge.sh/ that represents plugin registry generated from this PR.
- Open **index.json** file and check that each editor has an information about icon location
- Click on the icon path, the icon should be present
- All icons are located in https://pr-check-1885-che-plugin-registry.surge.sh/v3/images/

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
